### PR TITLE
Update Generic SDK Errors with Platform Errors

### PR DESCRIPTION
**Pull request by jumaallan**
Story: https://app.shortcut.com/smileid/story/12097/update-generic-errors-with-actual-platform-errors-in-mobile-sdks-v10

## Summary

This ticket changes the default error we show, and replace that with the actual platform code that we get from the api. We allow localization by allowing partners to overide the default empty strings defined on the sdk with the specific messages they would like to show their users

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

**Review by vanshg — COMMENTED** [review]
It might be better to have standard name format for the XML resource names, because then it would allow us to fetch them dynamically, like this: https://stackoverflow.com/questions/3476430/

That way we don't have to maintain a mapping, the resource would be calculated on the fly

File: lib/src/main/java/com/smileidentity/compose/biometric/OrchestratedBiometricKYCScreen.kt [hunk 1]
```diff
@@ -58,26 +58,19 @@ fun OrchestratedBiometricKYCScreen(
...
-                    uiState.errorMessageRes
-                        ?: R.string.si_biometric_kyc_processing_success_subtitle,
-                ),
+                successSubtitle = uiState.errorMessage.resolve().takeIf { it.isNotEmpty() }
+                    ?: stringResource(R.string.si_biometric_kyc_processing_success_subtitle),
                 successIcon = painterResource(R.drawable.si_processing_success),
                 errorTitle = stringResource(R.string.si_biometric_kyc_processing_error_subtitle),
-                errorSubtitle = errorSubtitle,
+                errorSubtitle = uiState.errorMessage.resolve().takeIf { it.isNotEmpty() }
+                    ?: stringResource(id = R.string.si_processing_error_subtitle),
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/compose/biometric/OrchestratedBiometricKYCScreen.kt:73]: You should use `derivedStateOf(uiState.errorMessage)` to compute and store this value. That way it is not recomputed on every recomposition, and only when `errorMessage` changes

File: lib/src/main/java/com/smileidentity/util/Util.kt [hunk 2]
```diff
@@ -259,30 +263,60 @@ fun getExceptionHandler(proxy: (Throwable) -> Unit) = CoroutineExceptionHandler
...
+                val resourceId = context.resources.getIdentifier(
+                    /* name = */
+                    resourceName,
+                    /* defType = */
+                    "string",
+                    /* defPackage = */
+                    context.packageName,
+                )
+
+                return stringResource(id = resourceId).takeIf { it.isNotEmpty() }
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/util/Util.kt:315]: What happens when the requested identifier does not exist? Will this fail gracefully/with an empty string?

File: lib/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -157,24 +157,24 @@
...
-    <string name="si_2212"></string>
-    <string name="si_2213_2413"></string>
-    <string name="si_2215"></string>
-    <string name="si_2216"></string>
-    <string name="si_2220"></string>
-    <string name="si_2221"></string>
-    <string name="si_2314"></string>
-    <string name="si_2413"></string>
-    <string name="si_2414"></string>
+    <string name="si_error_message_2201_2301_2401"></string>
```
Comment (vanshg) [inline lib/src/main/res/values/strings.xml:160]: How come we still have these multi-code error messages?

File: lib/src/main/java/com/smileidentity/compose/biometric/OrchestratedBiometricKYCScreen.kt (hunk 1 above)
Comment (jumaallan) [reply lib/src/main/java/com/smileidentity/compose/biometric/OrchestratedBiometricKYCScreen.kt:73]: `resolve()` needs to be inside the context of a Composable

File: lib/src/main/java/com/smileidentity/util/Util.kt (hunk 2 above)
Comment (jumaallan) [reply lib/src/main/java/com/smileidentity/util/Util.kt:315]: This will fail gracefully, then we try to get the message from the exception itself

File: lib/src/main/res/values/strings.xml (hunk 3 above)
Comment (jumaallan) [reply lib/src/main/res/values/strings.xml]: good catch, fixed

File: lib/src/main/java/com/smileidentity/compose/biometric/OrchestratedBiometricKYCScreen.kt (hunk 1 above)
Comment (vanshg) [reply lib/src/main/java/com/smileidentity/compose/biometric/OrchestratedBiometricKYCScreen.kt:73]: Makes sense

File: lib/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -157,17 +157,20 @@
...
+    <string name="si_error_message_2205"></string>
+    <string name="si_error_message_2405"></string>
     <string name="si_error_message_2207"></string>
     <string name="si_error_message_2208"></string>
     <string name="si_error_message_2209"></string>
     <string name="si_error_message_2210"></string>
     <string name="si_error_message_2211"></string>
     <string name="si_error_message_2212"></string>
-    <string name="si_error_message_2213_2413"></string>
+    <string name="si_error_message_2213"></string>
```
Comment (vanshg) [inline lib/src/main/res/values/strings.xml:173]:
```suggestion
    <string name="si_error_message_2213"></string>
    <string name="si_error_message_2214"></string>
```

File: lib/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -157,17 +157,20 @@
     <!--  SDK Errors  -->
     <string name="si_processing_error_subtitle">This could be because of image quality or internet connectivity</string>
     <string name="si_no_internet">Please check your internet connectivity</string>
-    <string name="si_error_message_2201_2301_2401"></string>
+    <string name="si_error_message_22011"></string>
```
Comment (vanshg) [inline lib/src/main/res/values/strings.xml:160]:
```suggestion
    <string name="si_error_message_2201"></string>
```

**Review by vanshg — APPROVED** [review]

File: lib/src/main/java/com/smileidentity/util/Util.kt (hunk 2 above)
Comment (vanshg) [reply lib/src/main/java/com/smileidentity/util/Util.kt:315]: This did not, in fact, fail gracefully -- it throws an exception when the resource isn't found, which takeIf does not handle
